lockbox-ios/Action/DataStoreAction.swift

```diff
@@ -0,0 +1,322 @@
+
+            webConfig.userContentController.add(self, name: JSCallbackFunction.OpenComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.InitializeComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.UnlockComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.LockComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.ListComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.GetComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.AddComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.UpdateComplete.rawValue)
+            webConfig.userContentController.add(self, name: JSCallbackFunction.DeleteComplete.rawValue)
+
+            webConfig.preferences.setValue(true, forKey: "allowFileAccessFromFileURLs")
+            webConfig.preferences.javaScriptEnabled = true
+
+            return webConfig
+        }
+    }
+
+    init(dataStoreName: String = "ds",
+         parser: ItemParser = Parser(),
+         dispatcher: Dispatcher = Dispatcher.shared) {
+        self.dataStoreName = dataStoreName
+        self.parser = parser
+        self.dispatcher = dispatcher
+        super.init()
+
+        self.webView = WebView(frame: .zero, configuration: self.webViewConfiguration)
+        self.webView.navigationDelegate = self
+
+        let baseUrl = URL(string: "file://\(Bundle.main.bundlePath)/lockbox-datastore/")!
+        let path = "file://\(Bundle.main.bundlePath)/lockbox-datastore/index.html"
+
+        self.webView.loadFileURL(URL(string: path)!, allowingReadAccessTo: baseUrl)
+    }
+
+    public func initialize(scopedKey: String, uid: String) {
+        self.initializeSubject
+                .take(1)
+                .subscribe(onNext: { [weak self] _ in
+                    self?.dispatcher.dispatch(action: DataStoreAction.initialized(initialized: true))
+                }, onError: { [weak self] error in
+                    self?.dispatcher.dispatch(action: ErrorAction(error: error))
+                    self?.initializeSubject = PublishSubject<Any>()
+                })
+                .disposed(by: self.disposeBag)
+
+        self._initialize(scopedKey: scopedKey, uid: uid)
+    }
+
+    public func updateInitialized() {
+        self._initialized()
+                .subscribe(onSuccess: { [weak self] initialized in
+                    self?.dispatcher.dispatch(action: DataStoreAction.initialized(initialized: initialized))
+                }, onError: { [weak self] error in
+                    self?.dispatcher.dispatch(action: ErrorAction(error: error))
+                })
+                .disposed(by: self.disposeBag)
+    }
+
+    public func unlock(scopedKey: String) {
+        self.unlockSubject
```

Comment:
The names correspond directly with the actual functions on the JavaScript datastore library. They could get more specific in Swift but I find it helpful to have them align so that it's easy to talk about what each function is doing in relation to the JavaScript getting executed.

Reply:
Ah, ok. Could you comment that they correspond to the JS datastore function, and calling them results in a call to that function. (even a link to that JSDoc of the function).